torch/_torch_docs.py

```diff
@@ -3508,25 +3508,164 @@ def merge_dicts(*dicts):
+
+.. math::
+    \begin{aligned}
+        {h_l}^2 f(x+h_r) - &{h_r}^2 f(x-h_l) + ({h_r}^2-{h_l}^2 ) f(x) = \\
+        &{h_l}^2 {h_r} f'(x) + {h_l}^2 {h_r}^3 \frac{f'''(x_r)}{6} + {h_r}^2
+        {h_l} f'(x) + {h_r}^2 {h_l}^3 \frac{f'''(x_l)}{6}
+    \end{aligned}
+
+
+after simple algebraic manipulations we get:
+
+.. math::
+    \begin{aligned}
+        {h_r} {h_l}^2 f'(x) + {h_r}^2 {h_l} f'(x) = &{h_l}^2 f(x+h_r) -
+            {h_r}^2 f(x-h_l) + ({h_r}^2-{h_l}^2 ) f(x) \\
+        &- {h_l}^2 {h_r}^3 \frac{f'''(x)}{6} + {h_r}^2 {h_l}^3
+            \frac{f'''(x)}{6}
+    \end{aligned}
+
+which concludes:
+
+.. math::
+    f'(x) =  \frac{ {h_l}^2 f(x+h_r) -{h_r}^2 f(x-h_l)
+            + ({h_r}^2-{h_l}^2 ) f(x) }{ {h_r} {h_l}^2 + {h_r}^2 {h_l} }
+            - {h_r}{h_l} \frac{ (f'''(x_r) + f'''(x_l))}{6}
+
+Because continuous functions are bounded in a closed domain and :math:`f \in C^3` , we can conclude
+when the multiplication :math:`h_r h_l \rightarrow 0` then we have:
+
+.. math::
+    f'(x) =  \frac{ {h_l}^2 f(x+h_r) -{h_r}^2 f(x-h_l)
+             + ({h_r}^2-{h_l}^2 ) f(x) }{ {h_r} {h_l}^2 + {h_r}^2 {h_l} }
+             - \mathcal{O}({h_r}{h_l}).
+
+
+Therefore, we can estimate :math:`f'(x)` by
+
+.. math::
+    f'(x) \approx  \frac{ {h_l}^2 f(x+h_r) -{h_r}^2 f(x-h_l)
+          + ({h_r}^2-{h_l}^2 ) f(x) }{ {h_r} {h_l}^2 + {h_r}^2 {h_l} }
+
+An interesting special case, happens when :math:`h_r = h_l=h` which gives us
+:math:`f'(x) \approx  \frac{  f(x+h) - f(x-h) }{ 2h}`.
+
+.. note::
+    We estimate the gradient of functions in complex domain
+    :math:`g : \mathbb{C}^n \rightarrow \mathbb{C}` in the same way.
 
 Args:
-    {input}
+    input (``Tensor``): the tensor that represents the values of the function
 
 Keyword args:
-    spacing (scalar, list of scalar, list of Tensor, optional): implicitly or explicitly represents
-    the coordinates the function is evaluated at
-    dim (int, list of int, optional): the dimension or dimensions to approximate the gradient over.
-    edge_order (int, optional): unsupported (must be equal to its default value which is 1.)
+    spacing (``scalar``, ``list of scalar``, ``list of Tensor``, optional): i) list of 1-dimensional Tensors
+        where each of them would represent explicit coordinates in the corresponding
+        directions.
+        ii) list of scalars where each scalar implicitly represents coordinates as the step size
+        in the corresponding direction as we discussed above.
+        iii) scalar, this is also an implicit representation of coordinates where step size in all
+        directions is the same.
+        iv) not provided, this case is equivalent to implicit representation where all coordinates
+        have a step size of 1.
 
-Example:
+    dim (``int``, ``list of int``, optional): the dimension or dimensions to approximate the gradient over.
+        When this parameter has not been provided, then we compute gradients in all dimensions.
```

Comment:
The first sentence for `dim` is good, but this second sentence needs a tweak.

"By default the partial gradient in each dimension is computed." 

Note that we don't want to use language like "we compute" because we're not computing anything, the function is.

Reply:
done ! Thanks for the suggestion !